cmd/namespace.go

```diff
@@ -1,14 +1,270 @@
+
+		clientConfig, err := config.LoadClientConfig(filePath)
+		if projectName == "" {
+			if dirPath == "" {
+				l.Info(fmt.Sprintf("Loading project name from client config in: %s", filePath))
+			}
+			if err != nil {
+				return err
+			}
+			projectName = clientConfig.Project.Name
+			l.Info(fmt.Sprintf("Using project name from client config: %s", projectName))
+		}
+		if namespaceName == "" {
+			return errors.New("namespace name is required")
+		}
+		if serverHost == "" {
+			if dirPath == "" {
+				l.Info(fmt.Sprintf("Loading service host from client config in: %s", filePath))
+			}
+			if err != nil {
+				return err
+			}
+			serverHost = clientConfig.Host
+			l.Info(fmt.Sprintf("Using server host from client config: %s", serverHost))
+		}
+
+		l.Info(fmt.Sprintf("Getting namespace [%s] in project [%s] from [%s]", namespaceName, projectName, serverHost))
+		namespace, err := getNamespace(projectName, namespaceName, serverHost)
+		if err != nil {
+			return err
+		}
+		result := stringifyNamespaceForNamespaceDescribe(namespace)
+		l.Info("Successfully getting namespace!")
+		l.Info(fmt.Sprintf("==============================\n%s", result))
+		return nil
+	}
+	cmd.Flags().StringVar(&dirPath, "dir", dirPath, "Directory where the Optimus client config resides")
+	cmd.Flags().StringVar(&serverHost, "server", serverHost, "Targeted server host, by default taking from client config")
+	cmd.Flags().StringVar(&projectName, "project-name", projectName, "Targeted project name, by default taking from client config")
+	cmd.Flags().StringVar(&namespaceName, "name", namespaceName, "Targeted namespace name, by default taking from client config")
+	return cmd
+}
+
+func stringifyNamespaceForNamespaceDescribe(namespace *config.Namespace) string {
+	output := fmt.Sprintf("name: %s\n", namespace.Name)
+	if len(namespace.Config) == 0 {
+		output += "config: {}"
+	} else {
+		output += "config:\n"
+		for key, value := range namespace.Config {
+			output += fmt.Sprintf("\t%s: %s", key, value)
+		}
+	}
+	return output
+}
+
+func getNamespace(projectName, namespaceName, serverHost string) (*config.Namespace, error) {
+	dialTimeoutCtx, dialCancel := context.WithTimeout(context.Background(), OptimusDialTimeout)
+	defer dialCancel()
+	requestTimeoutCtx, registerCancel := context.WithTimeout(context.Background(), deploymentTimeout)
+	defer registerCancel()
+
+	conn, err := createConnection(dialTimeoutCtx, serverHost)
+	if err != nil {
+		return nil, fmt.Errorf("failed creating connection to [%s]: %w", serverHost, err)
+	}
+
+	request := &pb.GetNamespaceRequest{
+		ProjectName:   projectName,
+		NamespaceName: namespaceName,
+	}
+
+	namespaceServiceClient := pb.NewNamespaceServiceClient(conn)
+	response, err := namespaceServiceClient.GetNamespace(requestTimeoutCtx, request)
+	if err != nil {
+		return nil, fmt.Errorf("unable to get namespace [%s]: %w", namespaceName, err)
+	}
+	return &config.Namespace{
```

Comment:
isn't config models to be used only for config loading, why can't we use the models.Namespacespec

Reply:
ignore